client/app/kubernetes/nodes/kubernetes-nodes.controller.js

```diff
@@ -0,0 +1,111 @@
+angular.module("managerApp").controller("KubernetesNodesCtrl", class KubernetesNodesCtrl {
+
+    constructor ($stateParams, $timeout, $translate, $uibModal, CloudMessage, Kubernetes, KUBERNETES) {
+        this.$stateParams = $stateParams;
+        this.$timeout = $timeout;
+        this.$translate = $translate;
+        this.$uibModal = $uibModal;
+        this.CloudMessage = CloudMessage;
+        this.Kubernetes = Kubernetes;
+        this.KUBERNETES = KUBERNETES;
+    }
+
+    $onInit () {
+        this.serviceName = this.$stateParams.serviceName;
+        this.loading = false;
+
+        this.loadMessages();
+        this.getNodes();
+        this.getPublicCloudProject();
+    }
+
+    loadMessages () {
+        this.CloudMessage.unSubscribe("paas.kube.nodes");
+        this.messageHandler = this.CloudMessage.subscribe("paas.kube.nodes", { onMessage: () => this.refreshMessages() });
+    }
+
+    refreshMessages () {
+        this.messages = this.messageHandler.getMessages();
+    }
+
+    getNodes () {
+        this.loading = true;
+        return this.Kubernetes.getNodes(this.serviceName)
+            .then(nodes => { this.nodes = nodes; })
+            .catch(() => this.CloudMessage.error(this.$translate.instant("kube_nodes_error")))
+            .finally(() => { this.loading = false; });
+    }
+
+    getAssociatedFlavor (node) {
+        return this.Kubernetes.getAssociatedInstance(node.projectId, node.instanceId)
+            .then(instance => _.set(node, "formattedFlavor", this.Kubernetes.formatFlavor(instance.flavor)))
+            .catch(() => {
+                _.set(node, "formattedFlavor", this.$translate.instant("kube_nodes_flavor_error"));
+            });
+    }
+
+    getPublicCloudProject () {
+        return this.Kubernetes.getAssociatedPublicCloudProjects(this.serviceName)
+            .then(projects => this.Kubernetes.getProject(_.first(projects).projectId))
+            .then(project => {
+                this.project = project;
+            })
+            .catch(() => {
+                this.CloudMessage.error(this.$translate.instant("kube_nodes_project_error"));
+            });
+    }
+
+    confirmNodeDeletion (nodeId) {
+        return this.$uibModal.open({
+            templateUrl: "app/kubernetes/nodes/delete/kubernetes-nodes-delete.html",
+            controller: "KubernetesNodesDeleteCtrl",
+            controllerAs: "$ctrl",
+            backdrop: "static",
+            resolve: {
+                nodeId () {
+                    return nodeId;
+                }
+            }
+        }).result
+            .then(() => {
+                this.displaySuccessMessage("kube_nodes_delete_success");
+                this.Kubernetes.resetNodesCache();
+                this.getNodes();
+            })
+            .catch(error => {
+                if (error) {
```

Comment:
You should display something even if the error is falsy

Reply:
`error` is undefined when the user dismisses the modal, I'm not sure an error message is needed then

Reply:
`error` is undefined when the user dismisses the modal, I'm not sure an error message is needed then